packages/ember-data/lib/serializers/rest-serializer.js

```diff
@@ -549,7 +548,7 @@ var RESTSerializer = JSONSerializer.extend({
     @return {String} the model's typeKey
   */
   typeForRoot: function(key) {
-    return camelize(singularize(key));
+    return normalizeTypeKey(key);
```

Comment:
Seems like we don't expose normalizeTypeKey as a public api/on the DS namespace, yet we use it in the REST serializer? I am kinda dubious of the same function being used both internally and externally.